test/subspace-test-service/src/mock.rs

```diff
@@ -178,6 +180,91 @@ impl MockPrimaryNode {
             .push(tx);
         rx
     }
+
+    /// Wait until a bundle that created by `author_key` at `slot` have submitted to the
+    /// transaction pool
+    pub async fn wait_for_system_bundle<SClient, SBlock>(
+        &mut self,
+        author_key: Sr25519Keyring,
+        system_domain_client: &SClient,
+    ) -> Result<(), Box<dyn Error>>
+    where
+        SBlock: BlockT,
+        SClient: HeaderBackend<SBlock>,
+    {
+        // Wait for the system executor catch up first, otherwise it may skip the slot
+        self.wait_system_executor_catch_up(system_domain_client)
+            .await?;
+
+        let slot = self.produce_slot().into();
+        let author_key = ExecutorPublicKey::unchecked_from(author_key.public().0);
+
+        // Check if bundle is already present at the transaction pool
+        for ready_tx in self.transaction_pool.ready() {
+            if is_bundle_match(ready_tx.data.encode().as_slice(), slot, &author_key)? {
+                return Ok(());
+            }
+        }
+
+        let mut import_tx_stream = self.transaction_pool.import_notification_stream();
+        while let Some(ready_tx_hash) = import_tx_stream.next().await {
```

Comment:
I think this is problematic in theory, the reason is same as the potential missing primary block notifications we had before. When the slot is produced, the executor will be notified instantly and the bundle production starts in the background. If the extrinsic containing the produced bundle is eventually submitted at some point between the last check and this subscription, the notification of that bundle will be lost.